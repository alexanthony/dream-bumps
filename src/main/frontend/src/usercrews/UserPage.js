import React, {PropTypes, Component} from 'react';
import { connect } from 'react-redux';
import {Tabs, Tab} from 'material-ui/Tabs';

import UserGenderTabContent from './UserGenderTabContent';

class UserPage extends Component {
  render() {
    return (
      <Tabs>
        <Tab label="Men">
          <UserGenderTabContent sex="male" userId={parseInt(this.props.params.userId, 10)} />
        </Tab>
        <Tab label="Women">
          <UserGenderTabContent sex="female" userId={parseInt(this.props.params.userId, 10)} />
        </Tab>
      </Tabs>
    );
  }
}

UserPage.propTypes = {
  params: PropTypes.shape({
    userId: PropTypes.string
  })
}

export default connect()(UserPage);
