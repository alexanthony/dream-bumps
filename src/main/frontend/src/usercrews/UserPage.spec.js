import React from 'react';
import { shallow } from 'enzyme';
import { UserPage } from './UserPage';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const component = shallow(
    <UserPage
      params={{userId: "5"}}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
  // We don't actually mount, so these don't get called...
  // expect(fetchUsers).toHaveBeenCalled();
  // expect(fetchUserCrewRankings).toHaveBeenCalled()
});
