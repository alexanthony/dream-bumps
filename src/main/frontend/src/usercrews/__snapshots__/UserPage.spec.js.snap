exports[`test Renders properly 1`] = `
<Tabs
  initialSelectedIndex={0}
  onChange={[Function]}>
  <Tab
    label="Men">
    <UserGenderTabContent
      sex="male"
      userId={5} />
  </Tab>
  <Tab
    label="Women">
    <UserGenderTabContent
      sex="female"
      userId={5} />
  </Tab>
</Tabs>
`;
